src/main.c

```diff
@@ -1232,6 +1235,24 @@ int main_2(int opt_vcd, int argc, char *argv[])
         }
     }
 
+#ifdef WAVE_USE_XID
+    if (GLOBALS->socket_xid) {
+        gboolean is_x11_display = FALSE;
+#ifdef GDK_WINDOWING_X11
+        {
+            GdkDisplay *display = gdk_display_get_default();
+            if (display && GDK_IS_X11_DISPLAY(display)) {
```

Comment:
`GDK_IS_X11_DISPLAY` already checks for `display != NULL`.
```suggestion
            if (GDK_IS_X11_DISPLAY(display)) {
```